api/v1beta2/awscluster_webhook.go

```diff
@@ -184,11 +214,13 @@ func (r *AWSCluster) validateControlPlaneLoadBalancerUpdate(oldlb, newlb *AWSLoa
 		// Block the update for Protocol :
 		// - if it was not set in old spec but added in new spec
 		// - if it was set in old spec but changed in new spec
-		if !cmp.Equal(newlb.HealthCheckProtocol, oldlb.HealthCheckProtocol) {
-			allErrs = append(allErrs,
-				field.Invalid(field.NewPath("spec", "controlPlaneLoadBalancer", "healthCheckProtocol"),
-					newlb.HealthCheckProtocol, "field is immutable once set"),
-			)
+		if oldlb.LoadBalancerType != LoadBalancerTypeClassic {
```

Comment:
nit: I think a comment might be helpful.

```suggestion
        // Allow update for a classic ELB, because an update is required
        // before upgrading to Kubernetes v1.31 or newer. See #5139. 
		if oldlb.LoadBalancerType != LoadBalancerTypeClassic {
```